examples/all-clusters-app/all-clusters-common/src/operational-state-delegate-impl.cpp

```diff
@@ -0,0 +1,400 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/EventLogging.h>
+#include <app/util/config.h>
+#include <operational-state-delegate-impl.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace OperationalState {
+
+using chip::Protocols::InteractionModel::Status;
+
+constexpr const char * kWasherPreSoak = "pre-soak";
+constexpr const char * kWasherRinse   = "rinse";
+constexpr const char * kWasherSpin    = "spin";
+/**
+ * Template class to present Enquriy Table
+ */
+template <typename T>
+struct EnquiryTable
+{
+    /**
+     * Endpoint Id
+     */
+    chip::EndpointId mEndpointId;
+    /**
+     * Cluster Id
+     */
+    chip::ClusterId mClusterId;
+    /**
+     * point of Array(Items)
+     */
+    T * pItems;
+    /**
+     * ArraySize of Array(Items)
+     */
+    size_t numOfItems;
+};
+
+/**
+ * Enquriy Table of Phase List
+ * Note: User Define
+ */
+GenericOperationalPhase opPhaseList[] = {
+    /**
+     * Phase List is null
+     */
+    GenericOperationalPhase(),
+    /**
+     * Phase List isn't null
+     */
+    // GenericOperationalPhase(kWasherPreSoak, strlen(kWasherPreSoak)),
+    // GenericOperationalPhase(kWasherRinse, strlen(kWasherRinse)),
+    // GenericOperationalPhase(kWasherSpin, strlen(kWasherSpin)),
+};
+
+/**
+ * Enquriy Table of Phase List corresponding to endpointId and clusterId
+ * Note: User Define
+ */
+constexpr EnquiryTable<GenericOperationalPhase> kPhaseListEnquiryTable[] = {
+    // EndpointId, ClusterId, Array of phaseList, ArraySize of phaseList
+    { 1, Clusters::OperationalState::Id, opPhaseList, ArraySize(opPhaseList) },
```

Comment:
the endpoint id here ?

Reply:
is that mandatory to have only this hardcoded endpoint here as 1 ?

Reply:
An example, I will use constexpr to instead it.